feat(docs): add forms and tables to general concept (#5319)
refinedev/refine: documentation/docs/guides-concepts/general-concepts/index.md

</Tabs>

### Forms <GuideBadge id="guides-concepts/forms" />

Refine provides a set of hooks to handle form state, validation, submission, autosave, and more. These hooks seamlessly integrate with popular UI libraries, making it easier to use their form components.

> - [React Hook Form](https://react-hook-form.com/) (for Headless, Chakra UI, Mantine users)- [Documentation](/packages/documentation/react-hook-form/useForm.md) - [Example](/examples/form/react-hook-form/useForm.md)

Reviewer: ```suggestion
- [React Hook Form](https://react-hook-form.com/) (for Headless, Material UI, and Chakra UI users)- [Documentation](/packages/documentation/react-hook-form/useForm.md) - [Example](/examples/form/react-hook-form/useForm.md)
```

> - [Ant Design Form](https://ant.design/components/form/#header) (for Ant Design users) - [Documentation](/api-reference/antd/hooks/form/useForm.md) - [Example](/examples/form/antd/useForm.md)

Reviewer: ```suggestion
- [Ant Design Form](https://ant.design/components/form/#header) - [Documentation](/api-reference/antd/hooks/form/useForm.md) - [Example](/examples/form/antd/useForm.md)
```

> - [Mantine Form](https://mantine.dev/form/use-form) (for Mantine users) - [Documentation](/api-reference/mantine/hooks/form/useForm.md) - [Example](/examples/form/mantine/useForm.md)

Reviewer: ```suggestion
- [Mantine Form](https://mantine.dev/form/use-form) - [Documentation](/api-reference/mantine/hooks/form/useForm.md) - [Example](/examples/form/mantine/useForm.md)
```
